### PR TITLE
feat: App containerization: dockerfile and settings

### PR DESCRIPTION
**Pull request by mcastellin**
With this commit I introduce a Dockerfile and additional settings to containerize the application. #3 

Scope:
Containerizing the application will require futher discussion to clarify:
- how we want to manage settings override for different environments
- how to replace sqllite with a Database container and docker-compose
- context for additional environments to deploy the application

Known Issues:
- containers uses a sqllite file as a database and with the current structure cannot be externalized with a docker volume
- missing documentation on how to build and use the docker image (will include after the introduction of the docker-compose)
- dockerfile does not include options to be used for app development

### CODE REVIEW

**Comment by mcastellin** [comment]
HI @XaviTorello @jbagot I am raising this PR so you can take a first look at the work being done with the dockerization.

I noticed we have only a single settings file for the project and, since I needed to have logging in the standart output for the container, I have created a new folder structure called `jobs/override` to contains environment-specific settings files.

I didn't move the current `jobs/settings.py` to maintain compatiliby but I wanted to run this setup by you since I'm not the world expert in Django. What are your thoughts?

**Review by XaviTorello — CHANGES_REQUESTED** [review]
Many thanks @mcastellin :)

Just a few things to discuss

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.8
```
Comment (XaviTorello) [inline Dockerfile:1]: Maybe we can go to the last stable 3.9

File: jobs/override/base.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from jobs.settings import *
```
Comment (XaviTorello) [inline jobs/override/base.py:1]: Maybe will be better to start packaging `jobs/settings/` instead of define an override. I think this will be simpler

**Review by XaviTorello — CHANGES_REQUESTED** [review]
Great, many thanks @mcastellin :beers:

Just some feedback to start the discussion :)

File: jobs/override/base.py (hunk 2 above)
Comment (mcastellin) [reply jobs/override/base.py:1]: Yes I agree, I didn't want to break existing environments, that's why I used a different name.. but if you're ok with it I will migrate to the settings folder

File: Dockerfile (hunk 1 above)
Comment (mcastellin) [reply Dockerfile:1]: will do

**Comment by mcastellin** [comment]
> /dev/null

not sure what you mean with this 🙂 can you elaborate?

**Comment by XaviTorello** [comment]
> > /dev/null
> 
> not sure what you mean with this 🙂 can you elaborate?

Sorry, pay no attention please. I can't delete this message, it was an unwanted submitted review xD

File: jobs/override/base.py (hunk 2 above)
Comment (XaviTorello) [reply jobs/override/base.py:1]: For me is OK, what do you think @jbagot?

File: Dockerfile (hunk 1 above)
Comment (mcastellin) [reply Dockerfile:1]: @XaviTorello djangorestframework-simplejwt==4.4.0 is not a valid version for python 3.9. The most recent available for this version is 4.3.0. Any thoughts?

Comment (mcastellin) [reply Dockerfile:1]: I am starting to get lots of errors with the application once upgraded to 3.9..

Comment (mcastellin) [reply Dockerfile:1]: I tried downgrade the pip package and although we can install all other requirements I started to get other kind of issues from the `base64` module. May I suggest to handle the upgrade to python 3.9 as a separate PR? 
I personally think it's more manageable to work with smaller scopes.

File: jobs/settings/base.py [hunk 3]
```diff
@@ -2,11 +2,11 @@
 from datetime import timedelta
 from distutils.util import strtobool
 
-BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
 SECRET_KEY = '@pzqp#x^+#(olu#wy(6=mi9&a8n+g&x#af#apn07@j=5oin=xb'
 
-DEBUG = strtobool(os.environ.get('APF_DEBUG', False))
+DEBUG = strtobool(os.environ.get('APF_DEBUG', 'False'))
```
Comment (mcastellin) [inline jobs/settings/base.py:9]: Quick note on this change, I noticed that has been introduced recently but it was failing the string to bool conversion for the boolean `False` so I changed the default value into a String

File: Dockerfile (hunk 1 above)
Comment (XaviTorello) [reply Dockerfile:1]: makes sense, maybe it's better to wait some days until the community upgrade their libs 💪 

many thanks @mcastellin

File: jobs/settings/base.py (hunk 3 above)
Comment (XaviTorello) [reply jobs/settings/base.py:9]: Perfect! ty

**Review by XaviTorello — APPROVED** [review]
Great! Many thanks!

File: jobs/override/base.py (hunk 2 above)
Comment (jbagot) [reply jobs/override/base.py]: Already done with the settings. I think with the new approach is better. Nice :)

**Review by jbagot — APPROVED** [review]
For me it's fine. With the new settings approach I'm confident. For now it's fine, later we will improve it if we need it.

**Pull request merged**